src/test/java/com/iota/iri/cache/CacheTest.java

```diff
@@ -0,0 +1,180 @@
+package com.iota.iri.cache;
+
+import com.iota.iri.cache.impl.CacheConfigurationImpl;
+import com.iota.iri.cache.impl.CacheImpl;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.HashFactory;
+import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.storage.Indexable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+public class CacheTest {
+
+    private static final String TEST_TRANSACTION_HASH = "TEST9TRANSACTION9TEST9TRANSACTION9TEST9TRANSACTION9TEST9TRANSACTION9TEST999999999";
+    private static final String TEST_TRANSACTION_HASH1 = "A9999TRANSACTION9TEST9TRANSACTION9TEST9TRANSACTION9TEST9TRANSACTION9TEST999999999";
+    private static final Hash hash = HashFactory.TRANSACTION.create(TEST_TRANSACTION_HASH);
+    private static final Hash hash1 = HashFactory.TRANSACTION.create(TEST_TRANSACTION_HASH1);
+
+    @Mock
+    private Cache<Indexable, TransactionViewModel> cache;
+
+    @Before
+    public void setUp() {
+        cache = Mockito.spy(new CacheImpl<>(new CacheConfigurationImpl()));
+    }
+
+    @After
+    public void tearDown() {
+        cache.clear();
+    }
+
+    @Test
+    public void shouldGetCachedValue() {
+        TransactionViewModel tvm = new TransactionViewModel(getTransaction(TEST_TRANSACTION_HASH), hash);
+        cache.put(hash, tvm);
+
+        Assert.assertEquals("Tranasction view models should be equal", tvm, cache.get(hash));
+    }
+
+    @Test
+    public void shouldGetAll() {
+        List<Indexable> keys = new ArrayList<Indexable>() {
+
+            {
+                add(hash);
+            }
+        };
+
+        TransactionViewModel tvm = new TransactionViewModel(getTransaction(TEST_TRANSACTION_HASH), hash);
+        Map<Indexable, TransactionViewModel> values = new HashMap<Indexable, TransactionViewModel>() {
+
+            {
+                put(hash, tvm);
+            }
+        };
+
+        cache.put(hash, tvm);
+
+        Assert.assertTrue("Values map should not be empty", values.size() > 0);
+        Assert.assertEquals("TVMs should be equals", tvm, cache.getAll(keys).get(hash));
+    }
+
+    @Test
+    public void shouldGetKeys() {
+        cache.put(hash, new TransactionViewModel(getTransaction(TEST_TRANSACTION_HASH), hash));
+        List<Indexable> keys = cache.getKeys();
+
+        Assert.assertTrue("Key list should not be empty", keys.size() > 0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4567264)